apps/passport-client/components/shared/PCDArgs.tsx

```diff
@@ -104,48 +153,75 @@ export function PCDArgs<T extends PCDPackage>({
 export function ArgInput<T extends PCDPackage, ArgName extends string>({
   arg,
   argName,
+  parentArgName,
   setArgs,
   defaultArg,
   hidden
 }: {
   arg: ArgsOf<T>[ArgName];
   argName: string;
+  parentArgName: string | undefined;
   setArgs: React.Dispatch<React.SetStateAction<ArgsOf<T>>>;
   defaultArg?: DisplayArg<typeof arg>;
   hidden?: boolean;
 }): JSX.Element | undefined {
+  // Go one level deeper in case the arg arises from a record.
   const setArg = React.useCallback(
     (value: (typeof arg)["value"]) => {
       setArgs((args) => ({
         ...args,
-        [argName]: {
-          ...args[argName],
-          value
-        }
+        ...(parentArgName !== undefined
+          ? {
+              [parentArgName]: {
+                ...args[parentArgName],
+                value: {
+                  ...args[parentArgName].value,
+                  [argName]: {
+                    ...args[parentArgName].value[argName],
+                    value
+                  }
+                }
+              }
+            }
+          : {
+              [argName]: {
+                ...args[argName],
+                value
+              }
+            })
```

Comment:
this is quite confusing and i anticipate will be difficult to maintain unless we break it up somehow